### ISSUE TITLE
onPremise requirements

### ISSUE THREAD

**Issue by dderichswei** [HewlettPackard/ezdemo#3]
With onPremise deployments like Baremetal, oVirt, VMware, HyperV we also need to support existing infrastructure components like AD/LDAP/PROXY,DNS/NTP...., right?

**Comment by erdincka** [comment]
I'd go with the most common scenario; assume there is DHCP/DNS, use AD/LDAP with optional selection (deploy your own or provide the server and DNs) and use public NTP (assuming internet access possible).

**Comment by erdincka** [comment]
Making everything as option/selection will make the scripts complicated, instead I'd try to keep these as variables so making them options becomes easier later.

**Comment by dderichswei** [comment]
I understand that we want to make it as simple as possible, on the other hand we also need a way to customize the different environments and parameters.

At the moment I have no idea how to implement this easily, ex. I found some hardcoded information like LDAP in Ansible! 
I would recommend that we outsource these things in a separate config folder/config file and make it configurable for the deployment if required.

**Comment by erdincka** [comment]
I totally agree to make these parameterised. Let me know what is the best way and I can work on that. Or if you already had these separated, we can replace them in place.

**Comment by erdincka** [comment]
I am testing a way to provide additional parameters directly into the ansible inventory file, and it works for VMWare deployment now. We can expand this to all targets. 
So the playbooks can check if a variable exist (such as http_proxy) and then run the necessary tasks.
